Add Max-Forwards Header (#20028)
mdn/content: files/en-us/web/http/headers/max-forward/index.md

browser-compat: http.headers.Max-Forwards
---
{{HTTPSidebar}}

The **`Max-Forwards`** request HTTP header is used with the [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method to limit the number of nodes (usually proxies) that request goes through. Its value is an integer value indicating the _maximum amount_ of nodes it must visit. At each node, the value is decremented and the `TRACE` request is forwarded to the next node, until the destination is reached, or the received value of `Max-Forwards` is zero. The request is then sent back, except for some headers, as the body of a `200 OK` response.

If the `Max-Forwards` header is not present in a `TRACE` request, a node will assume that there is no maximum number of forwards.

<table class="properties">
  <tbody>
    <tr>
      <th scope="row">Header type</th>
      <td>{{Glossary("Request header")}}</td>
    </tr>
    <tr>
      <th scope="row">{{Glossary("Forbidden header name")}}</th>
      <td>no</td>
    </tr>
  </tbody>
</table>

## Syntax

```http
Max-Forwards: <integer>
```
## Examples

```http
Max-Forwards: 0
Max-Forwards: 10
```

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

> - The HTTP [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method

Reviewer: ```suggestion
- The HTTP [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method

```

---
title: Max-Forwards
slug: Web/HTTP/Headers/Accept
tags:
  - HTTP
  - HTTP Header
  - Reference
  - Request header
browser-compat: http.headers.Accept
---
{{HTTPSidebar}}

> The **`Max-Forwards`** request HTTP header is used with the [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method to limit the number of nodes (usually proxies) that request goes through. Its value is an integer value indicating the _maximum amount_ of nodes it must visit. At each node, the value is decremented and the `TRACE` request is forwarded to the next node, until the destination is reached, or the received value of `Max-Forwards` is zero. The request is then forwarded back, except for some headers, as the body of a `200 OK` response.

Reviewer: ```suggestion
The **`Max-Forwards`** request HTTP header is used with the [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method to limit the number of nodes (usually proxies) that request goes through. Its value is an integer value indicating the _maximum amount_ of nodes it must visit. At each node, the value is decremented and the `TRACE` request is forwarded to the next node, until the destination is reached, or the received value of `Max-Forwards` is zero. The request is then sent back, except for some headers, as the body of a `200 OK` response.

```

If the `Max-Forwards` header is not present in a `TRACE` request, a node will assume that there is no maximum number of forwards.

<table class="properties">
  <tbody>
    <tr>
      <th scope="row">Header type</th>
      <td>{{Glossary("Request header")}}</td>
    </tr>
    <tr>
      <th scope="row">{{Glossary("Forbidden header name")}}</th>
      <td>no</td>
    </tr>
  </tbody>
</table>

## Syntax

```http
Max-Forwards: <integer>

> 

Reviewer: ```suggestion
```

---
title: Max-Forwards

> slug: Web/HTTP/Headers/Accept

Reviewer: ```suggestion
slug: Web/HTTP/Headers/Max-Forwards
```

tags:
  - HTTP
  - HTTP Header
  - Reference
  - Request header

> browser-compat: http.headers.Accept

Reviewer: ```suggestion
browser-compat: http.headers.Max-Forwards
```

---
{{HTTPSidebar}}

The **`Max-Forwards`** request HTTP header is used with the [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method to limit the number of nodes (usually proxies) that request goes through. Its value is an integer value indicating the _maximum amount_ of nodes it must visit. At each node, the value is decremented and the `TRACE` request is forwarded to the next node, until the destination is reached, or the received value of `Max-Forwards` is zero. The request is then sent back, except for some headers, as the body of a `200 OK` response.

> 

Reviewer: ```suggestion

```

---
title: Max-Forwards
slug: Web/HTTP/Headers/Accept
tags:
  - HTTP
  - HTTP Header
  - Reference
  - Request header
browser-compat: http.headers.Accept
---
{{HTTPSidebar}}

> The **`Max-Forwards`** request HTTP header is used only with the TRACE method,to limit the number of proxies that request goes through.A Integer Value is given to these Header.If a TRACE Request is received with the Max-Forwards method,each hop should decrement it and then forward the request to next hop.If the received value is zero (and so cannot be deincremented further), the node becomes authortative for the given request, instead of the origin server.If the Max-Forwards Header is missing in a TRACE Request ,assume that there is no maximum number of forwards.

Reviewer: ```suggestion
The **`Max-Forwards`** request HTTP header is used with the [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method to limit the number of nodes (usually proxies) that request goes through. Its value is an integer value indicating the _maximum amount_ of nodes it must visit. At each node, the value is decremented and the `TRACE` request is forwarded to the next node, until the destination is reached, or the received value of `Max-Forwards` is zero. The request is then forwarded back, except for some headers, as the body of a `200 OK` response.

If the `Max-Forwards` header is not present in a `TRACE` request, a node will assume that there is no maximum number of forwards.
```

<table class="properties">
  <tbody>
    <tr>
      <th scope="row">Header type</th>
      <td>{{Glossary("Request header")}}</td>
    </tr>
    <tr>
      <th scope="row">{{Glossary("Forbidden header name")}}</th>
      <td>no</td>
    </tr>
  </tbody>
</table>

## Syntax

```http

> Max-Forwards: <Integer-Value>

Reviewer: ```suggestion
Max-Forwards: <integer>

```

```
## Examples

```http
Max-Forwards: 0

> 

Reviewer: ```suggestion
```

Max-Forwards: 10

> 

Reviewer: ```suggestion
```

```

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

> - HTTP [content negotiation](/en-US/docs/Web/HTTP/Content_negotiation)
> - [List of default Accept values](/en-US/docs/Web/HTTP/Content_negotiation/List_of_default_Accept_values)
> - A header with the result of the content negotiation: {{HTTPHeader("Content-Type")}}
> - Other similar headers: {{HTTPHeader("TE")}}, {{HTTPHeader("Accept-Encoding")}}, {{HTTPHeader("Accept-Language")}}

Reviewer: ```suggestion
- The HTTP [`TRACE`](/en-US/docs/Web/HTTP/Methods/TRACE) method
```
